Incorrect error formatting for generic attribute missing errors

**Version Used**: 5fcefbef

**Steps to Reproduce**:

```cs
[SomeAttribute<int>] // error CS0246: The type or namespace name 'SomeAttribute<>Attribute' could not be found (are you missing a using directive or an assembly reference?)
void Foo() {}
```
[Sharplab](https://sharplab.io/#v2:C4LghgzsA0AmIGoA+BtAygewLYFMCCwwATgJYBGArsDgDwkB2wAfALoCwAUAAIAsABADEMGABQBKPgG8AvkA)

**Diagnostic Id**: CS0246

**Expected Behavior**:

`The type or namespace name 'SomeAttributeAttribute<>' could not be found (are you missing a using directive or an assembly reference?)`

**Actual Behavior**:

`The type or namespace name 'SomeAttribute<>Attribute' could not be found (are you missing a using directive or an assembly reference?)`

--- a/src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs
+++ b/src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs
@@ -2512,8 +2512,10 @@ namespace Microsoft.CodeAnalysis.CSharp
             // for attributes, suggest both, but not for verbatim name
             if (options.IsAttributeTypeLookup() && !options.IsVerbatimNameAttributeTypeLookup())
             {
-                // just recurse one level, so cheat and OR verbatim name option :)
-                NotFound(where, simpleName, arity, whereText + "Attribute", diagnostics, aliasOpt, qualifierOpt, options | LookupOptions.VerbatimNameAttributeTypeOnly);
+                // Check for generic attribute
+                string expectedDiagnostic = arity > 0 ? $"{simpleName}Attribute<>" : $"{simpleName}Attribute";
+
+                NotFound(where, simpleName, arity, expectedDiagnostic, diagnostics, aliasOpt, qualifierOpt, options | LookupOptions.VerbatimNameAttributeTypeOnly);
             }
 
             if ((object)qualifierOpt != null)

--- a/src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs
+++ b/src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs
@@ -2512,10 +2512,9 @@ namespace Microsoft.CodeAnalysis.CSharp
             // for attributes, suggest both, but not for verbatim name
             if (options.IsAttributeTypeLookup() && !options.IsVerbatimNameAttributeTypeLookup())
             {
-                // Check for generic attribute
-                string expectedDiagnostic = arity > 0 ? $"{simpleName}Attribute<>" : $"{simpleName}Attribute";
+                string attributeName = arity > 0 ? $"{simpleName}Attribute<>" : $"{simpleName}Attribute";
 
-                NotFound(where, simpleName, arity, expectedDiagnostic, diagnostics, aliasOpt, qualifierOpt, options | LookupOptions.VerbatimNameAttributeTypeOnly);
+                NotFound(where, simpleName, arity, attributeName, diagnostics, aliasOpt, qualifierOpt, options | LookupOptions.VerbatimNameAttributeTypeOnly);
             }
 
             if ((object)qualifierOpt != null)

--- a/src/Compilers/CSharp/Test/Symbol/Symbols/SymbolErrorTests.cs
+++ b/src/Compilers/CSharp/Test/Symbol/Symbols/SymbolErrorTests.cs
@@ -3476,7 +3476,7 @@ class BAttribute : System.Attribute { }
             var text =
 @"class C
 {
-    [SomeAttribute<int>] // Error should occur here
+    [SomeAttribute<int>]
     static void Foo()
     {
     }
@@ -3484,7 +3484,11 @@ class BAttribute : System.Attribute { }
 ";
             CreateCompilation(text).
                 VerifyDiagnostics(
+                // (3,6): error CS0246: The type or namespace name 'SomeAttributeAttribute<>' could not be found (are you missing a using directive or an assembly reference?)
+                //      [SomeAttribute<int>]
                 Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "SomeAttribute<int>").WithArguments("SomeAttributeAttribute<>"),
+                // (3,6): error CS0246: The type or namespace name 'SomeAttribute<>' could not be found (are you missing a using directive or an assembly reference?)
+                //      [SomeAttribute<int>]
                 Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "SomeAttribute<int>").WithArguments("SomeAttribute<>")
                 );
         }

--- a/src/Compilers/CSharp/Test/Symbol/Symbols/SymbolErrorTests.cs
+++ b/src/Compilers/CSharp/Test/Symbol/Symbols/SymbolErrorTests.cs
@@ -3477,7 +3477,7 @@ class BAttribute : System.Attribute { }
 @"class C
 {
     [SomeAttribute<int>]
-    static void Foo()
+    static void M()
     {
     }
 }
@@ -3490,6 +3490,29 @@ class BAttribute : System.Attribute { }
                 // (3,6): error CS0246: The type or namespace name 'SomeAttribute<>' could not be found (are you missing a using directive or an assembly reference?)
                 //      [SomeAttribute<int>]
                 Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "SomeAttribute<int>").WithArguments("SomeAttribute<>")
+                );
+        }
+
+        [Fact, WorkItem(69700, "https://github.com/dotnet/roslyn/issues/69700")]
+        public void CS0246ERR_SingleTypeNameNotFound08()
+        {
+            var text =
+@"class C
+{
+    [Some<int>]
+    static void M()
+    {
+    }
+}
+";
+            CreateCompilation(text).
+                VerifyDiagnostics(
+                // (3,6): error CS0246: The type or namespace name 'SomeAttribute<>' could not be found (are you missing a using directive or an assembly reference?)
+                //      [Some<int>]
+                Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "Some<int>").WithArguments("SomeAttribute<>"),
+                // (3,6): error CS0246: The type or namespace name 'Some<>' could not be found (are you missing a using directive or an assembly reference?)
+                //      [Some<int>]
+                Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "Some<int>").WithArguments("Some<>")
                 );
         }
 

--- a/src/Compilers/CSharp/Test/Symbol/Symbols/SymbolErrorTests.cs
+++ b/src/Compilers/CSharp/Test/Symbol/Symbols/SymbolErrorTests.cs
@@ -3469,7 +3469,8 @@ class BAttribute : System.Attribute { }
                 Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "Nada").WithArguments("Nada")
                 );
         }
-        
+
+        [WorkItem(69700, "https://github.com/dotnet/roslyn/issues/69700")]
         [Fact]
         public void CS0246ERR_SingleTypeNameNotFound07()
         {

--- a/src/Compilers/CSharp/Test/Symbol/Symbols/SymbolErrorTests.cs
+++ b/src/Compilers/CSharp/Test/Symbol/Symbols/SymbolErrorTests.cs
@@ -3470,8 +3470,7 @@ class BAttribute : System.Attribute { }
                 );
         }
 
-        [WorkItem(69700, "https://github.com/dotnet/roslyn/issues/69700")]
-        [Fact]
+        [Fact, WorkItem(69700, "https://github.com/dotnet/roslyn/issues/69700")]
         public void CS0246ERR_SingleTypeNameNotFound07()
         {
             var text =

--- a/src/Compilers/CSharp/Test/Symbol/Symbols/SymbolErrorTests.cs
+++ b/src/Compilers/CSharp/Test/Symbol/Symbols/SymbolErrorTests.cs
@@ -3469,6 +3469,25 @@ class BAttribute : System.Attribute { }
                 Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "Nada").WithArguments("Nada")
                 );
         }
+        
+        [Fact]
+        public void CS0246ERR_SingleTypeNameNotFound07()
+        {
+            var text =
+@"class C
+{
+    [SomeAttribute<int>] // Error should occur here
+    static void Foo()
+    {
+    }
+}
+";
+            CreateCompilation(text).
+                VerifyDiagnostics(
+                Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "SomeAttribute<int>").WithArguments("SomeAttributeAttribute<>"),
+                Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "SomeAttribute<int>").WithArguments("SomeAttribute<>")
+                );
+        }
 
         [Fact]
         public void CS0249ERR_OverrideFinalizeDeprecated()

